app/src/main/java/io/stormbird/wallet/entity/SignaturePair.java

```diff
@@ -153,11 +153,15 @@ public SignaturePair(String qrMessage, String timeMessage, String contractAddr)
         //Now convert sig back to Byte
         signature = sigBi.toByteArray();
 
-        if (signature.length == 64)
+        if (signature.length < 65)
         {
+            int offset = 65 - signature.length;
             byte[] sigCopy = new byte[65];
-            System.arraycopy(signature, 0, sigCopy, 1, 64);
-            sigCopy[0] = 0;
+            System.arraycopy(signature, 0, sigCopy, offset, 65-offset);
+            for (int i = 0; i < offset; i++)
+            {
+                sigCopy[0] = 0;
```

Comment:
Maybe should be:

    sigCopy[i] = 0;

?

Reply:
Yes it should be. Thanks :) and thanks for looking, even if I'm not sure how you got here. Yes the buffer probably does initialise implicitly to zero, but for a signature it's best to be completely sure.

Be careful of making yourself too useful we don't want you to burn yourself out.